pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -144,3 +146,12 @@ func validateMetric(annotations map[string]string) *apis.FieldError {
 	}
 	return nil
 }
+
+func validateInitialScale(allowInitScaleZero bool, annotations map[string]string) *apis.FieldError {
+	if initialScale, ok := annotations[InitialScaleAnnotationKey]; ok {
+		if !allowInitScaleZero && initialScale == "0" {
```

Comment:
This should probably validate that the value is actually an integer. That'd make the == "0" check also more typesafe, as it could be == 0 in that case.